cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/graph/scopes/Scope.kt

```diff
@@ -227,6 +239,20 @@ sealed class Scope(
             list += value
         }
     }
+
+    fun firstParentOrNull(predicate: (Scope) -> Boolean): Scope? {
```

Comment:
`firstParentScopeOrNull` to not confuse it with the `astParent`?

Reply:
Was my original idea as well. I can change it.